### PR TITLE
Retention settings are not available in create topic - Kafka Janitor

### PR DESCRIPTION
**Pull request by SEQUOIIA**
Story details: https://app.clubhouse.io/devex/story/2110

### CODE REVIEW

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 137 lines, deletes 20 lines of code in 10 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 140 lines, deletes 20 lines of code in 10 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 290 lines, deletes 32 lines of code in 15 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>3 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

File: src/Specifications/TestDoubles/TikaTopicsClientSpy.cs [hunk 1]
```diff
@@ -1,38 +1,54 @@
...
+        public readonly List<TopicDescription> TopicDescriptions = new List<TopicDescription>();
         public Task<TopicDescription> DescribeAsync(string topicName)
         {
-            Topics.Find(v => v == topicName);
+            var topicDescription = TopicDescriptions
+                .Single(t => 
+                    t.name == topicName
+                );
 
-            return Task.FromResult(new TopicDescription());
```
Comment (SEQUOIIA) [inline src/Specifications/TestDoubles/TikaTopicsClientSpy.cs]: Ah, I assume this was the one you mentioned @kim-lindhard-dfds ?

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 291 lines, deletes 33 lines of code in 15 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>3 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

**Pull request merged**